### PR TITLE
feat(web): remember task listing display preferences

### PR DESCRIPTION
**Pull request by carlosflorencio**
The task listing now retains each device's preferred view while fixing **All Workflows** on single-workflow boards. Users can optionally reveal richer List-row context, and that detail preference follows them across signed-in devices.

## Important Changes

- Store the last Kanban, Pipeline, or List selection locally per device; phones render Kanban for a saved Pipeline choice without overwriting it.
- Add the portable **Show task details** List preference and richer repository, description, pull-request, session, parent, review, and archive context.

## Validation

- `make fmt`
- `make typecheck`
- `make test`
- `make lint`
- Focused Playwright: 6 desktop Chromium scenarios and 2 Pixel 5 scenarios.
- `node --test scripts/validate-public-docs.test.mjs` and `node scripts/validate-public-docs.mjs`

## Checklist

- [ ] I have performed a self-review of my code.
- [ ] I have manually tested my changes and they work as expected.
- [ ] My changes have tests that cover the new functionality and edge cases.
- [ ] If my change touches UI files (`apps/web/`), I have added or updated Playwright e2e tests in `apps/web/e2e/` and verified them with `make test-e2e`.
- [ ] I checked whether this affects public docs in `docs/public/**` and updated them or noted why no docs change is needed.

<a href="https://cubic.dev/pr/kdlbs/kandev/pull/1944?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

## Screenshots

Desktop List view with rich task details

Mobile List view with rich task details

### CODE REVIEW

File: apps/web/lib/types/backend.ts [hunk 1]
```diff
@@ -351,6 +351,7 @@ export type UserSettingsUpdatedPayload = {
   user_id: string;
   workspace_id: string;
   kanban_view_mode?: string;
+  tasks_list_show_details?: boolean;
```
[the remark this answers is not in the document]
Comment (carlosflorencio) [reply apps/web/lib/types/backend.ts:373]: Fixed in d6f9a4b. Live user-settings events now map tasks_list_show_details and preserve the current value when omitted; a handler regression covers both cases.

File: apps/web/lib/task-listing/view-preference.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    return null;
+  }
+}
+
+export function resolveTaskListingView(
+  storedView: TaskListingView | null,
+  legacyKanbanViewMode: string | null | undefined,
+): TaskListingView {
+  if (storedView) return storedView;
+  return legacyKanbanViewMode === "graph2" ? "pipeline" : DEFAULT_TASK_LISTING_VIEW;
```
[the remark this answers is not in the document]
Comment (carlosflorencio) [reply apps/web/lib/task-listing/view-preference.ts:25]: Fixed in d6f9a4b. An invalid stored device preference now falls back to Kanban; legacy Pipeline migration applies only when the device key is absent.

File: apps/web/app/page-client.tsx [hunk 3]
```diff
@@ -26,6 +31,12 @@ export function PageClient({ initialTaskId, initialSessionId }: PageClientProps)
     setBoardKey((prev) => prev + 1);
   };
 
+  useEffect(() => {
+    if (shouldRestoreHomeTaskListingView(preferredView, initialTaskId, initialSessionId)) {
+      router.replace("/tasks");
+    }
+  }, [initialSessionId, initialTaskId, preferredView, router]);
```
[the remark this answers is not in the document]
Comment (carlosflorencio) [reply apps/web/app/page-client.tsx]: Fixed in d6f9a4b. Home-to-List restoration now carries the resolved workspace through linkToTasks, with a PageClient regression test.

File: apps/web/app/tasks/rich-task-row-details.ts [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    (a, b) => a.position - b.position,
+  )) {
+    if (seenRepositoryIds.has(taskRepository.repository_id)) continue;
+    seenRepositoryIds.add(taskRepository.repository_id);
+    const repository = repositoriesById.get(taskRepository.repository_id);
+    if (!repository) continue;
+    const label = repositorySlug(repository);
+    if (seenRepositoryLabels.has(label)) continue;
+    seenRepositoryLabels.add(label);
+    repositoryLabels.push(label);
```
[the remark this answers is not in the document]
Comment (carlosflorencio) [reply apps/web/app/tasks/rich-task-row-details.ts:41]: Fixed in d6f9a4b. Rich repository labels now use provider owner/name or repository name only, never local paths; a local-path-only regression covers omission.

File: apps/web/hooks/use-task-listing-view.ts [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    if (userSettings.kanbanViewMode === kanbanViewMode) return;
+    setUserSettings({ ...userSettings, kanbanViewMode, loaded: true });
+  }, [effectiveView, setUserSettings, userSettings]);
+
+  const setView = useCallback((view: TaskListingView) => {
+    setStoredTaskListingView(view);
+  }, []);
+
+  return { preferredView, effectiveView, setView };
+}
```
[the remark this answers is not in the document]
Comment (carlosflorencio) [reply apps/web/hooks/use-task-listing-view.ts:47]: Fixed in d6f9a4b. Added hook-level coverage for custom and storage updates plus the mobile Pipeline effective-Kanban synchronization.

File: docs/specs/INDEX.md [hunk 6]
```diff
@@ -128,6 +128,7 @@ Subscription quota tracking and per-agent cheap-model profile routing.
 | [task-layout-profiles](ui/task-layout-profiles.md) | draft |
 | [agent-message-comments](ui/agent-message-comments.md) | shipped |
 | [external-vcs-file-links](ui/external-vcs-file-links.md) | shipped |
+| [task-listing-display-preferences](ui/task-listing-display-preferences.md) | draft |
```
[the remark this answers is not in the document]
Comment (carlosflorencio) [reply docs/specs/INDEX.md]: Fixed in d6f9a4b. The specs index now marks task-listing-display-preferences as shipped.

File: apps/web/hooks/use-task-listing-view.test.tsx [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+
+    act(() => {
+      window.localStorage.setItem(TASK_LISTING_VIEW_STORAGE_KEY, '"pipeline"');
+      window.dispatchEvent(new CustomEvent(TASK_LISTING_VIEW_CHANGE_EVENT));
+    });
+    expect(result.current.preferredView).toBe("pipeline");
+
+    act(() => {
+      window.localStorage.setItem(TASK_LISTING_VIEW_STORAGE_KEY, '"list"');
+      window.dispatchEvent(new StorageEvent("storage", { key: TASK_LISTING_VIEW_STORAGE_KEY }));
```
[the remark this answers is not in the document]
Comment (carlosflorencio) [reply apps/web/hooks/use-task-listing-view.test.tsx]: Fixed in cfdac55. The storage fixture now creates StorageEvent with only its type and defines key explicitly before dispatching.

File: apps/web/app/page-client.test.tsx [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+    });
+  });
+
+  it("does not restore List while opening a task", async () => {
+    render(<PageClient workspaceId="workspace-1" initialTaskId="task-1" />);
+
+    await waitFor(() => {
+      expect(replaceMock).not.toHaveBeenCalled();
+    });
+  });
```
[the remark this answers is not in the document]
Comment (carlosflorencio) [reply apps/web/app/page-client.test.tsx:40]: Fixed in cfdac55. Added the session-only deep-link case to confirm List restoration does not redirect it.

File: apps/web/hooks/use-task-listing-view.test.tsx [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+
+    act(() => {
+      window.localStorage.setItem(TASK_LISTING_VIEW_STORAGE_KEY, '"pipeline"');
+      window.dispatchEvent(new CustomEvent(TASK_LISTING_VIEW_CHANGE_EVENT));
+    });
+    expect(result.current.preferredView).toBe("pipeline");
+
+    act(() => {
+      window.localStorage.setItem(TASK_LISTING_VIEW_STORAGE_KEY, '"list"');
+      const storageEvent = new StorageEvent("storage");
```
[the remark this answers is not in the document]
Comment (carlosflorencio) [reply apps/web/hooks/use-task-listing-view.test.tsx]: Fixed in 146a09e22. The subscription only depends on the event type, so the fixture now dispatches a generic Event("storage") and defines the storage key used by the scenario. This preserves the cross-tab notification behavior without invoking CodeQL's incompatible StorageEvent constructor signature.

**Comment by carlosflorencio** [comment]
Already covered by apps/backend/internal/backendapp/helpers_test.go: the boot payload test asserts taskPendingAction uses the exact JSON key and has the expected permission value for both active Kanban and multi-workflow snapshot tasks.
